ChameleonMiniGUI/Json/MifareUltralightModel.cs

```diff
@@ -37,32 +37,148 @@ public MifareUltralightCardInfo Card
         public override byte[] ToByteArray()
         {
             var stba = new Func<string, byte[]>(MifareClassicModel.StringToByteArray);
-            return stba(info.Version)
-                .Concat(stba(info.TBO_0))
-                .Concat(stba(info.Tearing))
-                .Concat(stba(info.Pack))
-                .Concat(stba(info.TBO_1))
-                .Concat(stba(info.Signature))
-                .Concat(Blocks.SelectMany(bytes => bytes))
-                .ToArray();
+            if (info.Counter0 != null && info.Counter1 != null && info.Counter2 != null) {
+                // Save as new header dump format with counters
+                byte[] pages = { (byte) (Blocks.Length - 1) };
+                return stba(info.Version)
+                    .Concat(stba(info.TBO_0))
+                    .Concat(stba(info.TBO_1))
+                    .Concat(pages)
+                    .Concat(stba(info.Signature))
+                    .Concat(stba(info.Counter0)).Concat(stba(info.Tearing0))
+                    .Concat(stba(info.Counter1)).Concat(stba(info.Tearing1))
+                    .Concat(stba(info.Counter2)).Concat(stba(info.Tearing2))
+                    .Concat(Blocks.SelectMany(bytes => bytes))
+                    .ToArray();
+            }
+            else if (info.Tearing0 != null && info.Tearing1 != null && info.Tearing2 != null)
```

Comment:
Is this suppose to a detection of format used before?  or`?  Kind of unclear why you do these different choices.

Reply:
removed